src/app/submit-button.tsx

```diff
@@ -1,31 +1,65 @@
-export const SubmitButton = () => {
+import { gql, useMutation } from '@apollo/client';
+import React from 'react';
 
+interface ISubmitButtonProps {
+    inputsData: {
+        email: string;
+        password: string;
+    }
+    inputError : {
+       email: boolean;
+       password: boolean; 
+    }
+    onClick: () => void;
+    validate: boolean;
 
-    return (
-        <>
-            <button type="button" onClick={inputValidator}>Entrar</button>
-        </>
-    );
+}
 
-};
+const loginMutation = gql`
+    mutation Login($data: LoginInput!) {
+        login(data: $data) {
+            token
+        }
+    }
+`;
 
-const inputValidator = () => {
-    const emailValidator = () => {
-        const email = document.querySelector("#email-input") as HTMLInputElement;
-        return email.value.match(/^\S+@\S+\.\S+$/);
-    };
-    const passwordValidator = () => {
-        const password = document.querySelector("#password-input") as HTMLInputElement;
-        // This regex finds matches if the input.value has a length < 7 and doesn't contain at least
-        // a digit and one letter.
-        return password.value.match(/^(.{0,6}|[^0-9]*|[^a-z])$/);
-    };
+export const SubmitButton = (props: ISubmitButtonProps) => {
```

Comment:
Está perfeito! Mas costumamos tipar os React components usando o React.FC


```suggestion
export const SubmitButton: React.FC<ISubmitButtonProps> = (props) => {
```